### PR TITLE
Fix an issue where MLD requests are silently dropped

### PR DESCRIPTION
**Pull request by kderda**
This PR resolves problems with handling MLD requests and the corresponding test case as both were not working as expected.

### CODE REVIEW

File: subsys/net/ip/ipv6_mld.c [hunk 1]
```diff
@@ -124,16 +124,19 @@ static int mld_create_packet(struct net_pkt *pkt, uint16_t count)
 
 static int mld_send(struct net_pkt *pkt)
 {
+	int res;
```
Comment (jukkar) [inline subsys/net/ip/ipv6_mld.c:127]: Please use `ret` as a variable name (just to be consistent with other functions in this file)

**Review by jukkar — COMMENTED** [review]
Small nit about variable name.
Should we change the commit ordering, so first the change, and then the test?

**Comment by konradderda** [comment]
> Should we change the commit ordering, so first the change, and then the test?
I just kept them in a sequence I was resolving the issue, to me both options sound good.

**Comment by jukkar** [comment]
The ordering could affect bisectability, so usually with other PRs we introduce first the change, and then the tests for it.

**Comment by konradderda** [comment]
@jukkar I have swapped the commits and fixed issue where tests with `CONFIG_NET_TC_THREAD_PREEMPTIVE=y` were failing.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**